### PR TITLE
Fixes #6508: Error 500 when sending multiple files to inventory endpoint

### PR DESCRIPTION
**Pull request by fanf**
The correction is only to cleany handle 0 and more than 1 case for the multipart. Most changes are just spaces, and the factorisation-out of the error message. 

I had to also change from DefaultMultipartHttpServletRequest to HttpServletRequest and then pattern match on the type to cleanly manage the 0 case.

### CODE REVIEW

File: inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala [hunk 1]
```diff
@@ -87,62 +88,77 @@ class FusionReportEndpoint(
...
-            case _ =>
-              val msg = "The report is empty, not saving anything."
+  def onSubmit(request: HttpServletRequest) = {
+    def defaultBadAnswer(reason: String) = new ResponseEntity(s"""${reason} sent. You have to POST a request with exactly one file in attachment (with 'content-disposition': file)
+                               |For example, for curl, use: curl -F "file=@path/to/file"
+                               |""".stripMargin, HttpStatus.PRECONDITION_FAILED)
+
+
+    request match {
+      case mutlipart: DefaultMultipartHttpServletRequest =>
```
Comment (ncharles) [inline inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala]: there's a typo there - could you write multipart instead ?

File: inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala [hunk 2]
```diff
@@ -87,62 +88,77 @@ class FusionReportEndpoint(
...
+
+        val files = mutlipart.getFileMap.values
+
+        if(files.size <= 0) {
+          defaultBadAnswer("No report")
```
Comment (ncharles) [inline inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala]: report, or inventory ?

File: inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala [hunk 3]
```diff
@@ -87,62 +88,77 @@ class FusionReportEndpoint(
...
+        } else if(files.size == 1) {
+          val reportFile = files.toSeq(0)
+          //copy the session file somewhere where it won't be deleted on that method return
+          logger.info("New input report: '%s'".format(reportFile.getOriginalFilename))
```
Comment (ncharles) [inline inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala]: report, or inventory

File: inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala [hunk 4]
```diff
@@ -87,62 +88,77 @@ class FusionReportEndpoint(
...
+          //val reportFile = copyFileToTempDir(f)
+
+          var in : InputStream = null
+          logger.trace("Start post parsing report '%s'".format(reportFile.getOriginalFilename))
```
Comment (ncharles) [inline inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala]: same

File: inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala [hunk 5]
```diff
@@ -87,62 +88,77 @@ class FusionReportEndpoint(
...
+          try {
+            in = reportFile.getInputStream
+            val start = System.currentTimeMillis
+
+            (unmarshaller.fromXml(reportFile.getName,in) ?~! "Can't parse the input report, aborting") match {
```
Comment (ncharles) [inline inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala]: same

Comment (fanf) [inline inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/FusionReportEndpoint.scala]: We used to call that "report", but inventory makes much more sense.

**Pull request merged**